gnd/src/main/java/com/google/android/gnd/repository/DataRepository.java

```diff
@@ -248,13 +260,21 @@ public Completable saveFeature(Feature feature) {
         .andThen(dataSyncWorkManager.enqueueSyncWorker(feature.getId()));
   }
 
+  private Maybe<Project> loadLastActiveProject() {
+    String projectId = localValueStore.getLastActiveProjectId();
+    if (isOffline()) {
```

Comment:
A safer and potentially cleaner way of doing this is just falling back to local db on the error case, and setting an explicit timeout as I do for loading records. This way we also catch the case when the OS thinks the network is available, but in practice it isn't (common in environments w/intermittent connectivity).

Reply:
(in either case we'll need to handle the error case)